src/System.Private.CoreLib/shared/System/IO/Stream.cs

```diff
@@ -189,6 +189,65 @@ private int GetCopyBufferSize()
             return bufferSize;
         }
 
+        public virtual void CopyTo(ReadOnlySpanAction<byte, object> callback, object state, int bufferSize) =>
+            CopyTo(new WriteCallbackStream(callback, state), bufferSize);
+
+        public virtual Task CopyToAsync(Func<object, ReadOnlyMemory<byte>, CancellationToken, Task> callback, object state, int bufferSize, CancellationToken cancellationToken) =>
+            CopyToAsync(new WriteCallbackStream(callback, state), bufferSize, cancellationToken);
+
+        private sealed class WriteCallbackStream : Stream
+        {
+            private readonly ReadOnlySpanAction<byte, object>? _action;
+            private readonly Func<object, ReadOnlyMemory<byte>, CancellationToken, Task>? _func;
+            private readonly object _state;
+
+            public WriteCallbackStream(ReadOnlySpanAction<byte, object> action, object state)
+            {
+                _action = action ?? throw new ArgumentNullException(nameof(action));
```

Comment:
Do argument validation in the public entry point, not in the ctor. It makes the call stack look better and helps ensure we use the correct argument name if we need to throw an exception.

Reply:
I agree. Fixed.